vnext/include/Shared/fixed_string.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#if (_MSVC_LANG >= 202002L)
+#define CONSTEVAL consteval
+#else
+#define CONSTEVAL constexpr
+#endif
+namespace details {
+
+#ifdef FNV_HASH_WIDTH_64
+using hash_t = uint64_t;
+inline constexpr size_t FNV_offset_basis = 14695981039346656037ULL;
+inline constexpr size_t FNV_prime = 1099511628211ULL;
+#else
+using hash_t = uint32_t;
+inline constexpr hash_t FNV_offset_basis = 2166136261U;
+inline constexpr hash_t FNV_prime = 16777619U;
+#endif
+
+[[nodiscard]] inline CONSTEVAL hash_t Fnv1a_append_bytes(
+    hash_t _Val,
+    const char *_First,
+    const size_t _Count) noexcept { // accumulate range [_First, _First + _Count) into partial FNV-1a hash _Val
+  for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
+    _Val ^= static_cast<hash_t>(_First[_Idx]);
+    _Val *= FNV_prime;
+  }
+
+  return _Val;
+}
+
+_NODISCARD inline constexpr hash_t Fnv1a_append_bytes_constexpr(
+    hash_t _Val,
+    const char *_First,
+    const size_t _Count) noexcept { // accumulate range [_First, _First + _Count) into partial FNV-1a hash _Val
+  for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
+    _Val ^= static_cast<hash_t>(_First[_Idx]);
+    _Val *= FNV_prime;
+  }
+
+  return _Val;
+}
+
+inline hash_t hash(const char *str) noexcept {
+  return Fnv1a_append_bytes_constexpr(FNV_offset_basis, str, strlen(str));
+}
+
+template <size_t N>
+struct fixed_string {
```

Comment:
>fixed_string [](http://example.com/codeflow?start=7&length=12)

Could we Pascal casing for the names and files as we do in the rest of the code? It is better to keep common naming conventions.